google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingService.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.gax.core.ApiFuture;
+import com.google.api.gax.core.ApiFutureCallback;
+import com.google.api.gax.core.ApiFutures;
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.Logging.WriteOption;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Set;
+
+public class LoggingService {
```

Comment:
I've been thinking about this name & structure, and here are my thoughts:

* Calling this `LoggingService` is confusing because then it's not clear how `Logging` and `LoggingService` are supposed to be related - they are both for calling the Logging service, and neither of them are "local" services.
* It seems like all of this functionality could just be rolled into `Logging`/`LoggingImpl`. There is one potential point of confusion, the fact that `Logging` has `write` and `writeAsync`. Maybe that distinction should just be removed, and the sync/async should only be controlled by the synchronicity setting, not by which function is called. The `publish` function on this class could become the new `write` function in `Logging`.